### PR TITLE
ENH less pr_json

### PR DESCRIPTION
**Pull request by beckermr**
This pr makes the pr json a lot smaller. ~We will have to migrate the repo first.~ Also we should look at the PR keys and try and make sure it is complete. The initial estimate is that we will see a 50x reduction in size. 

to do
 - [x] tests
 - [x] agree on complete set of keys
 - [x] doc strings

### CODE REVIEW

File: conda_forge_tick/git_utils.py [hunk 1]
```diff
@@ -40,13 +42,52 @@
...
+    "html_url": None,
+    "created_at": None,
+    "updated_at": None,
+    "merged_at": None,
+    "state": None,
+    "mergeable_state": None,
+    "labels": None,
+    "head": {"ref": None},
+    "base": {"repo": {"name": None}},
+}
```
Comment (beckermr) [inline conda_forge_tick/git_utils.py:57]: @CJ-Wright Can you look at the github api docs (https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#get-a-pull-request) and see what you think should be here?

Comment (CJ-Wright) [reply conda_forge_tick/git_utils.py:60]: I think this is ok, I need to look at `by-the-numbers` to see if we need any more keys.

File: conda_forge_tick/update_prs.py [hunk 2]
```diff
@@ -119,32 +61,27 @@ def _update_pr(update_function, dry_run, gx):
...
-
-        for f in as_completed(futures):
+                if pr_json and not pr_json["state"] == "closed":
+                    future = pool.submit(update_function, ghctx, pr_json, gh, dry_run)
+                    futures[future] = (node_id, i, pr_json)
+
+        for f in tqdm.tqdm(
+            as_completed(futures),
+            total=len(futures),
+            desc="gathering PR data",
```
Comment (beckermr) [inline conda_forge_tick/update_prs.py:71]:
```suggestion
            desc="gathering PR data",
            leave=False,
```

**Comment by beckermr** [comment]
OK @CJ-Wright. This is ready for review. We can merge it right away, though I do want to baby sit. I have tested the PR updating bits and they work fine. The other git operations I did not test since they have to be run live.

**Comment by beckermr** [comment]
There is a script to cut down all of the old PR json in this PR as well. Once the bot is working well with this change, I will run it to make sure the old PR json gets smaller too.

File: conda_forge_tick/git_utils.py [hunk 3]
```diff
@@ -259,6 +310,100 @@ def delete_branch(ctx: GithubContext, pr_json: LazyJson, dry_run: bool = False)
...
+        A dict-like object with the current PR information.
+    """
+    hdrs = {
+        "Authorization": f"token {ctx.github_password}",
+        "Accept": "application/vnd.github.v3+json",
+    }
+    if not force and "ETag" in pr_json:
+        hdrs["If-None-Match"] = pr_json["ETag"]
+
+    r = requests.get(
```
Comment (CJ-Wright) [inline conda_forge_tick/git_utils.py:389]: Does this handle the fake PRs that we write when we don't issue a PR?

**Review by CJ-Wright — COMMENTED** [review]
One question, but otherwise looks good!

File: conda_forge_tick/git_utils.py [hunk 4]
```diff
@@ -270,24 +415,47 @@ def refresh_pr(
     gh: Optional[github3.GitHub] = None,
     dry_run: bool = False,
 ) -> Optional[dict]:
-    gh = ensure_gh(ctx, gh)
     if not pr_json["state"] == "closed":
```
Comment (beckermr) [inline conda_forge_tick/git_utils.py:418]: See here @CJ-Wright

File: conda_forge_tick/git_utils.py (hunk 3 above)
Comment (beckermr) [reply conda_forge_tick/git_utils.py:389]: This is how the code worked previous too. Those fake PRs fed directly to github3 would raise an error as well.

File: conda_forge_tick/git_utils.py (hunk 4 above)
Comment (CJ-Wright) [reply conda_forge_tick/git_utils.py:418]: Thank you!

**Pull request merged**